src/btree/bt_cursor.c

```diff
@@ -1057,9 +1071,16 @@ retry:	if (positioned == POSITIONED)
 			 * and that's not correct.
 			 */
 			cbt->recno = cursor->recno;
-		} else
-			ret = __cursor_col_modify(
-			    session, cbt, WT_UPDATE_TOMBSTONE);
+		} else {
+			WT_ERR(__wt_cursor_valid(cbt, NULL, &valid));
+			if (!valid) {
+				if (!__cursor_fix_implicit(btree, cbt))
+					WT_ERR(WT_NOTFOUND);
+				cbt->recno = cursor->recno;
+			} else
+				ret = __cursor_col_modify(
+				    session, cbt, WT_UPDATE_TOMBSTONE);
+		}
```

Comment:
I think it's unfortunate that we have to repeat the `__cursor_fix_implicit` here.

Did you consider calling `__wt_cursor_valid` before checking `cbt->compare != 0`?  Or would that generate a prepare conflict in some case where `WT_NOTFOUND` is expected?